packages/vm/core/blocklog/blockinfo.go

```diff
@@ -28,8 +32,46 @@ type BlockInfo struct {
 	TotalRequests         uint16
 	NumSuccessfulRequests uint16 // which didn't panic
 	NumOffLedgerRequests  uint16
-	GasBurned             uint64     `bcs:"compact"`
-	GasFeeCharged         coin.Value `bcs:"compact"`
+	GasBurned             uint64            `bcs:"compact"`
+	GasFeeCharged         coin.Value        `bcs:"compact"`
+	Entropy               hashing.HashValue // since v1
+}
+
+func (bi *BlockInfo) MarshalBCS(e *bcs.Encoder) error {
```

Comment:
@chriscandless is this better?

For some values (e.g. timestamp) I'm still using Encode/Decode. Is there a better way?

Also, can you check that the encoding/decoding of schema version 0 is exactly the same? The old struct is this: https://github.com/iotaledger/wasp/blob/24e99c374a8ef976e14330734d1f3fe45acc7db7/packages/vm/core/blocklog/blockinfo.go#L22